backend/src/scraper/bestbuy_scraper.py

```diff
@@ -45,12 +48,112 @@ def _product_specs(product_id: str) -> Dict:
             }
 
     @staticmethod
-    def _product_reviews(product_id: str) -> Dict:
-        # TODO: Implement review scraping.
-        return {"reviews": {}}
+    def _product_reviews(product_id: str, product_name: str) -> Dict:
+        # Search for product by UPC.
+        params = {
+            "product_id": product_id,
+            "headers": {"User-Agent": "Mozilla/5.0 (X11; CrOS x86_64 12871.102.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.141 Safari/537.36"},
+        }
+        search = requests.get(f"https://www.bestbuy.ca/api/v2/json/search?query={params.get('product_id')}", headers = params.get('headers'))
+        results = search.json()
+
+        if results.get('total') == 0:
+            return {
+                "reviews": {
+                    "error": "No reviews exist for provided id.",
+                }
+            }
+        else:
+            # Search for reviews by product ID.
+            canada_sku = results.get('products')[0].get('sku')
+            search_sku = requests.get(f"https://www.bestbuy.ca/api/v2/json/reviews/{canada_sku}?page=1&pagesize=700&source=us", headers = params.get('headers'))
+            review_results = search_sku.json()
+
+            # Compare names of US and Canadian products using cosine similarity.
+            canada_name = results.get('products')[0].get('name')
+            name_1 = word_tokenize(product_name.lower())  
+            name_2 = word_tokenize(canada_name.lower())
+
+            sw = stopwords.words('english') 
+            ps = PorterStemmer()
+            l1 =[]
+            l2 =[] 
+            
+            name_1_set = {ps.stem(w) for w in name_1 if not w in sw}  
+            name_2_set = {ps.stem(w) for w in name_2 if not w in sw} 
+            rvector = name_1_set.union(name_2_set)
+            for w in rvector: 
+                if w in name_1_set: l1.append(1)
+                else: l1.append(0) 
+                if w in name_2_set: l2.append(1) 
+                else: l2.append(0) 
```

Comment:
nit (don't feel like you have to make this change, but just a formatting tip):

```suggestion
                l1.append(1) if w in name_1_set else l1.append(0) 
                l2.append(1) if w in name_2_set else l2.append(0) 
```